tests/bud.bats

```diff
@@ -246,23 +246,44 @@ _EOF
   # With explicit and full --platform, buildah should not warn.
   run_buildah build $WITH_POLICY_JSON --platform linux/amd64/v2 \
               -t source -f $containerfile
-  assert "$output" !~ "missing .* build argument" \
-         "With explicit --platform, buildah should not warn"
+  assert "$output" =~ "image platform \(linux/amd64\) does not match the expected platform" \
+         "With explicit --platform, buildah should warn about pulling difference in platform"
+  assert "$output" =~ "TARGETOS=linux" " --platform TARGETOS set correctly"
+  assert "$output" =~ "TARGETARCH=amd64" " --platform TARGETARCH set correctly"
+  assert "$output" =~ "TARGETVARIANT=" " --platform TARGETVARIANT set correctly"
```

Comment:
Is TARGETVARIANT not being set to "v2", per "--platform"?

Reply:
I think this reveals the fix just not being correct — the core issue is that the `buildDockerfilesOnce ` code setting `TARGET*` expects `SystemContext.*Choice` to never be `""`, and that’s not how the values are set.

Compare more in https://github.com/containers/buildah/pull/5543#pullrequestreview-2074705651